### PR TITLE
Add Map View to Browse Events

### LINKED ISSUE #68: Browse events in Map View: UI
## Link to User Stories
#16, User flow 4 in Figma

## Objective
- be able to view the _approximate_ location of all events on a Map
- achieve feature parity with the list view

## Acceptance Criteria
- [ ] make the map accessible from a floating action button on the "Browse Events" tab on the homescreen (with the ability to go back to the list using the same button)

### LINKED ISSUE #118: Browse events in Map View: ViewModel + Backend
## Link to User Stories
#16, User flow 4 in Figma

## Related Issues
#68

## Objective
- be able to view the _approximate_ location of all events on a Map
- achieve feature parity with the list view

## Acceptance Criteria
- [ ] create a Map.kt and MapViewModel.kt
- [ ] accept a list of Events and show their approximate location on the map

### PR DESCRIPTION
**Pull request by jgeaso1266**
## Objective
- be able to view the approximate location of all events on a Map
- achieve feature parity with the list view

## Key Changes
- [ ] Created `Map.kt` that holds google maps composable
- [ ] link the `EventsOverviewViewModel()` to the Map UI
- [ ] Added floating action button to home screen
- [ ] Added logic to show the correct list view (map or list)
- [ ] Added UI test

closes #68 
closes #118

### CODE REVIEW

**Review by christinehahaa — APPROVED** [review]
LGTM! Keep it up :)

File: app/.idea/.name
Comment (Nath-Assa) [inline app/.idea/.name:1]: Not sure what this file does.

File: app/src/main/java/com/github/se/eventradar/ui/home/Home.kt [hunk 1]
```diff
@@ -159,6 +177,23 @@ fun HomeScreen(
               start.linkTo(parent.start)
               end.linkTo(parent.end)
             })
+
+    var viewToggleIcon =
+        if (viewToggleBrowseIndex == 0) Icons.Default.Place else Icons.AutoMirrored.Filled.List
+    FloatingActionButton(
+        onClick = {
+          viewToggleBrowseIndex = if (viewToggleBrowseIndex == 0) 1 else 0
+          viewToggleIcon =
```
Comment (Nath-Assa) [inline app/src/main/java/com/github/se/eventradar/ui/home/Home.kt:186]: nit: The var "viewToggleIcon" may not be necessary. Compose's reactivity means that the UI will redraw whenever state changes, so you can define the Icon directly based on viewToggleBrowseIndex when the icon is being drawn, not when the button is clicked. Like this: 
 Icon(
        imageVector = if (viewToggleBrowseIndex == 0) Icons.Default.Place else Icons.AutoMirrored.Filled.List,
        contentDescription = null
    )

**Review by Nath-Assa — APPROVED** [review]
LGTM! just a few nits

File: app/src/main/java/com/github/se/eventradar/ui/home/Home.kt (hunk 1 above)
Comment (jgeaso1266) [reply app/src/main/java/com/github/se/eventradar/ui/home/Home.kt:186]: True, I decided to write it this way mostly to be a little more verbose in the code. Also, I didn't want to have wait until the view was recomposed for the icon to change.

**Pull request merged**